File: leetcode/lc-us/cpp/2595.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> evenOddBit(int n) {
        vector<int> res(2);
        int i = 0;
        while (n) {
            if (n & 1) res[i & 1]++;
            i++, n >>= 1;
        }
        return res;
    }
};

File: leetcode/lc-us/rust/2595.rs
impl Solution {
    pub fn even_odd_bit(n: i32) -> Vec<i32> {
        let mut res = vec![0; 2];
        let mut i = 0;
        let mut n = n;
        while n > 0 {
            if (n & 1) == 1 {
                if (i & 1) == 1 {
                    res[1] += 1;
                } else {
                    res[0] += 1;
                }
            }
            n >>= 1;
            i += 1;
        }
        return res;
    }
}